tests/periph_i2c/tests/base_if.py

```diff
@@ -0,0 +1,173 @@
+import serial
+import logging
+import errno
+import os
+import time
+import serial.tools.list_ports
+
+
+class IfParams:
+    cmd = ''
+    result = ''
+    msg = ''
+    data = ''
+
+
+class BaseIf:
+    __READ_REG_CMD = "rr"
+    __WRITE_REG_CMD = "wr"
+    __EXECUTE_CMD = "ex"
+    __RESET_CMD = "mcu_rst"
+    __SUCCESS = '0'
+    __RESULT_SUCCESS = 'Success'
+    __RESULT_ERROR = 'Error'
+    __RESULT_TIMEOUT = 'Timeout'
+
+    def __init__(self, port='/dev/ttyACM0', baud=115200):
+        self.connect(port, baud)
+
+    def get_port(self):
+        if (self.__dev.isOpen()):
+            return self.__dev.port
+        return []
+
+    def autoconnect(self, expected_val, fxn, used_com=[]):
+        found_connection = False
+        comlist = serial.tools.list_ports.comports()
+        connected = []
+        logging.debug("Autoconnecting")
+        for element in comlist:
+            if (element.device not in used_com):
+                connected.append(element.device)
+        for port in connected:
+            logging.debug("Port: " + port)
+            self.connect(port)
+            ret = fxn().data
+            try:
+                logging.debug("ID rx: %s" % ret)
```

Comment:
the *new* python way for string formatting would look like this:

```
 logging.debug("ID rx: {}".format(ret))
```

simply replace every `%s` or `%d` and such by `{}`,
that way you don't need to care about variable types (which is not very pythonic anyway)